vcd/resource_vcd_vapp_vm.go

```diff
@@ -1789,6 +1984,12 @@ func resourceVcdVAppVmUpdateExecute(d *schema.ResourceData, meta interface{}, ex
 			if err != nil {
 				return diag.Errorf(errorCompletingTask, err)
 			}
+
+			err = vm.Refresh()
+			if err != nil {
+				return diag.Errorf("error refreshing VM: %s", err)
+			}
+
```

Comment:
If we do a refresh here, shouldn't we also do it in the end of `if customizationNeeded {`?

Reply:
Yes, moved the refresh outside of the if statements, right before the `if enter_bios_setup` branch